channels/jaicp/src/main/kotlin/com/justai/jaicf/channel/jaicp/execution/ThreadPoolRequestExecutor.kt

```diff
@@ -0,0 +1,52 @@
+package com.justai.jaicf.channel.jaicp.execution
+
+import com.justai.jaicf.channel.BotChannel
+import com.justai.jaicf.channel.http.asHttpBotRequest
+import com.justai.jaicf.channel.jaicp.JaicpCompatibleAsyncBotChannel
+import com.justai.jaicf.channel.jaicp.JaicpCompatibleBotChannel
+import com.justai.jaicf.channel.jaicp.JaicpMDC
+import com.justai.jaicf.channel.jaicp.channels.JaicpNativeBotChannel
+import com.justai.jaicf.channel.jaicp.dto.JaicpBotRequest
+import com.justai.jaicf.channel.jaicp.dto.JaicpBotResponse
+import com.justai.jaicf.channel.jaicp.processCompatible
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.async
+import kotlinx.coroutines.runBlocking
+import java.util.concurrent.Executors
+import kotlin.coroutines.CoroutineContext
+
+internal class ThreadPoolRequestExecutor(nThreads: Int) : CoroutineScope {
```

Comment:
Do we really need a whole new class just for proxying functions' invocations to executor?

Reply:
Single Responsibility. A single class to contain coroutineContext and processing methods with sync/async facade.